File: Warehousing.Data/Entities/ProductRecipe.cs
namespace Warehousing.Data.Entities
{
    public class ProductRecipe : BaseClass
    {
        public int Id { get; set; }
        
        public int ParentProductId { get; set; }  // The finished product
        public Product ParentProduct { get; set; } = null!;
        
        public int ComponentProductId { get; set; }  // The ingredient/component
        public Product ComponentProduct { get; set; } = null!;
        
        public decimal Quantity { get; set; }  // How much of component needed
        public bool IsActive { get; set; } = true;
        public string? Notes { get; set; } = string.Empty;
    }
}

File: Warehousing.UI/src/app/admin/models/productRecipe.ts
import { Product } from "./product";

export interface ProductRecipe {
    id: number;
    parentProductId: number;
    parentProduct: Product;
    componentProductId: number;
    componentProduct: Product;
    quantity: number;
    isActive: boolean;
    notes: string | null;
    createdAt: string | null;
    createdBy: string | null;
    updatedAt: string | null;
    updatedBy: string | null;
}

export interface ProductRecipeDto {
    id: number | null;
    parentProductId: number;
    componentProductId: number;
    quantity: number;
    isActive: boolean;
    notes: string | null;
}



